### DISCUSSION TITLE
柱状图问题

### DISCUSSION THREAD

**Discussion by zhiyou1201** [antvis/G2#5711] [Q&A]
如何分组展示呢
我的数据格式是这样的
[
            {
                "date_times": "2023-10-30", // 日期
                "ps_sale_total_price": 2377.6, // 配送金额
                "cost_money": 635, // 成本金额
                "lirun_price": 343 // 利润金额
            },
            {
                "date_times": "2023-10-31",
                "ps_sale_total_price": 0,
                "cost_money": 0,
                "lirun_price": 0
            },
            {
                "date_times": "2023-11-01",
                "ps_sale_total_price": 0,
                "cost_money": 0,
                "lirun_price": 0
            },
            {
                "date_times": "2023-11-02",
                "ps_sale_total_price": 0,
                "cost_money": 0,
                "lirun_price": 0
            }
]
每个时间 展示三个 配送金额 成本金额 利润金额
网站的例子我看数据格式跟我不一样  我无法搞定跟我这样数据格式的怎么写
![image](https://github.com/antvis/G2/assets/75651528/976556ee-956d-4db3-9835-40115db52bd1)

最终效果要这样的
![image](https://github.com/antvis/G2/assets/75651528/647948cc-be8c-455d-addd-e4189d2376b3)

**Comment by pearmini** [comment]
用 [fold transform](https://g2.antv.antgroup.com/spec/data/fold)

**Reply by zhiyou1201** [reply]
实在是搞不懂啊  能帮忙写一小部分吗  感谢 感谢 感谢了

**Reply by pearmini** [reply]
类似这样：

```js
chart
  .interval()
  .data({
    type: "inline",
    value: this.zhuData.RiProfit,
    transform: [
      {
        type: "fold",
        fields: [
          "date_times",
          "ps_sale_total_price",
          "cost_money",
          "lirun_price",
        ],
        key: 'key',
        value: 'value',
      },
    ],
  })
  .transform({ type: "sortX", by: "y", reverse: true, slice: 6 })
  .transform({ type: "dodgeX" })
  .encode("x", "key")
  .encode("y", "value")
  .encode("color", "ps_sale_total_price")
  .scale("y", { nice: true })
  .axis("y", { labelFormatter: "~s" });
```

**Comment by zhiyou1201** [comment]
实在是搞不懂啊 能帮忙写一小部分吗 感谢 感谢 感谢了

**Comment by zhiyou1201** [comment]
你回答的跟我问的没什么关系啊 我不是要转换数据  我是要怎么才能实现效果

**Comment by zhiyou1201** [comment]
我使用了fold transform实现不了功能啊  帮帮忙吧  怎么搞啊
